### ISSUE TITLE
Make ConsoleFormatter's context configurable

### ISSUE THREAD

**Issue by pblesi** [mattbrictson/airbrussh#130]
Right now Airbrussh, provides an `Airbrussh::Rake::Context`. This Context is used in the `ConsoleFormatter` and provides the following interface: `current_task_name`, `register_new_command`, and `position`. In the context of Capistrano, this context is enabled. When a new task is started, the current task changes and the history is reset. Outside the context of Capistrano, when using SSHKit directly, `current_task_name` always returns `nil` and `position` becomes an incrementing number for the entirety of execution.

It would be useful if `Airbrussh::Rake::Context` was abstracted so that a context could be set using configuration. This way SSHKit executors that do not use Capistrano/Rake (e.x. https://github.com/braintree/runbook) could define their own implementations of `current_task_name`, `register_new_command`, and `position`.

**Comment by mattbrictson** [comment]
Thanks for the suggestion! I'm interested in this idea.

What do you think the configuration syntax for this would look like? Is it something the user will put in their `Capfile` or `deploy.rb`?

**Comment by mattbrictson** [comment]
> Is it something the user will put in their `Capfile` or `deploy.rb`?

Actually, I guess the answer is "neither" since this is intended for users who are _not_ using Capistrano. In that case could you provide an example of what this configuration would look like with plain SSHKit or something like runbook?

**Comment by pblesi** [comment]
Right. So this is how Runbook currently initializes Airbrussh through SSHKit:

https://github.com/braintree/runbook/blob/4a0f0770a8a2a7be135cf13ee435d981b5975a06/lib/runbook/configuration.rb#L92-L96

If Airbrussh took a `context` option as part of its format options, then this would be a way to configure the context. This class would default to `Airbrussh::Rake::Context`. Another class could be set that adheres to the specified interface (`current_task_name`, `register_new_command`, and `position`). 

I believe the configuration could analogously be applied to the `set :format_options` interface used by capistrano: 

https://github.com/mattbrictson/airbrussh#configuration

The syntax in the Airbrussh::Formatter initializer case would be:

```
      ssh_kit.output = Airbrussh::Formatter.new(
        $stdout,
        banner: nil,
        command_output: true,
        context: Runbook::Airbrussh::Context
      )
```

And in the Capistrano case would be:

```
set :format_options, color: false, truncate: 80, context: CustomAirbrusshContext
```

I'd be happy to throw together a PR

**Comment by mattbrictson** [comment]
Makes sense. Thanks for the writeup.

> I'd be happy to throw together a PR

Yes please! 🙏

**Comment by mattbrictson** [comment]
Fixed via #131
